calculation/seistech_calc/seistech_calc/plots/plt_plotting.py

```diff
@@ -549,3 +618,610 @@ def plot_uhs_branches(
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_spectra(
+    gms_result: gms.GMSResult,
+    save_file: Path = None,
+):
+    """Plot of the pSA values of the realisations and
+     selected ground motions and the median, 16th,
+     and 84th percentile of the GCIM
+
+    Parameters
+    ----------
+    gms_result: gms.GMSResult
+    save_file: Path, optional
+    """
+    (
+        gcim_df,
+        realisations_df,
+        selected_gms_df,
+    ) = _prepare_gms_spectra(gms_result)
+
+    plt.figure(figsize=(20, 9))
+
+    for label, cur_gcim in gcim_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_gcim.index,
+            cur_gcim.values,
+            color="red",
+            linestyle="solid" if label == "median" else "dashdot",
+            label=GCIM_LABEL[label],
+        )
+
+    for cur_ix, cur_rel in realisations_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_rel.index.values,
+            cur_rel.values,
+            color="blue",
+            linestyle="solid",
+            label="Realisations" if cur_ix == 0 else None,
+            linewidth=0.4,
+        )
+
+    for cur_ix, cur_rel in selected_gms_df.iloc[:, 1:].iterrows():
+        plt.plot(
+            cur_rel.index.values,
+            cur_rel.values,
+            color="black",
+            linestyle="solid",
+            label="Selected Ground Motions" if cur_ix == 0 else None,
+            linewidth=0.4,
+        )
+
+    plt.xscale("log")
+    plt.yscale("log")
+    plt.xlabel("Period, T (s)")
+    plt.ylabel("Spectral acceleration, SA (g)")
+    plt.title("Pseudo acceleration response spectra")
+    plt.legend()
+
+    if save_file is not None:
+        plt.savefig(save_file)
+        plt.close()
+    else:
+        plt.show()
+
+
+def plot_gms_disagg_distribution(
+    contribution: List,
+    distribution: List,
+    gms_result: gms.GMSResult,
+    metadata_key: str,
+    cs_param_bounds: gms.CausalParamBounds = None,
+    save_file: Path = None,
+):
+    """CDF plots for the selected GMs and
+    all ruptures (disaggregation) for Magnitude
+    and Distance (Rrup)
+
+    Also shows causal bounding parameters if specified
+
+    Parameters
+    ----------
+    contribution: List
+        Cumulative sum of the contribution of the ruptures
+        corresponding to the distribution
+    distribution: List
+        List of either Mw or Rrup in an ascending order
+    gms_result: gms.GMSResult
+    metadata_key: str
+    cs_param_bounds: gms.CausalParamBounds, optional
+    save_file: Path, optional
+    """
+    selected_gms_metadata = gms_result.selected_gms_metdata_df.to_dict(orient="list")
+    bounds = _get_causal_params_bounds(cs_param_bounds)
+    bounds_y_range = [0, 1]
+    # To achieve Empirical distribution function with matplotlib's step plotting
+    copied_metadata = selected_gms_metadata[metadata_key][:]
```

Comment:
I suggest changing these lines to the following (only changed the first one, and renamed variable)
```
    metadata_values = list(gms_result.selected_gms_metdata_df.loc[:, metadata_key].values)
    metadata_values.append(min(copied_metadata))
    metadata_values.sort()

```
that way you can remove the line at 998

Reply:
Good to practice more of DataFrame then converting them to dictionary :+1: